scalameta/contrib/src/main/scala/scala/meta/contrib/DocToken.scala

```diff
@@ -0,0 +1,246 @@
+package scala.meta.contrib
+
+import scala.meta.contrib.DocToken.DocKind
+
+/**
+  * Represents a scaladoc line.
+  */
+case class DocToken(kind: DocKind, name: Option[String], body: String) {
+
+  override def toString: String = {
+    (name match {
+      case Some(n) => s"$kind(name=$n, body=$body)"
+      case _ => s"$kind($body)"
+    }).replaceAll("\n", " ")
+  }
+
+  /**
+    * Appends a this [[DocToken#body]] to the input [[DocToken#body]].
+    */
+  def append(docToken: DocToken) : DocToken = copy(body = s"$body\n${docToken.body}")
+}
+
+/**
+  * Companion object containing the classes required for describing an ScalaDoc token.
+  *
+  * The available tokens and their documentation are obtained from:
+  * @see http://docs.scala-lang.org/overviews/scaladoc/for-library-authors.html
+  */
+object DocToken {
+
+  /**
+    * Helper [[DocToken]] apply method.
+    */
+  def apply(kind: DocKind, body: String): DocToken =
+    new DocToken(kind, None, body)
+
+  /**
+    * Helper apply method for named [[DocToken]].
+    */
+  def apply(kind: DocKind, name: String, body: String): DocToken =
+    new DocToken(kind, Option(name), body)
+
+  /**
+    * Trait used for representing each type of documentation label.
+    */
+  sealed trait DocKind {
```

Comment:
How about `sealed abstract class Kind(val label: String)`? That'd make it possible to do `Param extends Kind("@param")`